crates/ruff_server/src/session/index/ruff_settings.rs

```diff
@@ -186,29 +205,52 @@ impl RuffSettingsIndex {
                     }
                 }
 
-                if let Some(pyproject) = settings_toml(&directory).ok().flatten() {
-                    let Ok(settings) = ruff_workspace::resolver::resolve_root_settings(
-                        &pyproject,
-                        Relativity::Parent,
-                        &EditorConfigurationTransformer(editor_settings, root),
-                    ) else {
-                        return WalkState::Continue;
-                    };
-                    index.write().unwrap().insert(
-                        directory,
-                        Arc::new(RuffSettings {
-                            path: Some(pyproject),
-                            file_resolver: settings.file_resolver,
-                            linter: settings.linter,
-                            formatter: settings.formatter,
-                        }),
-                    );
+                match settings_toml(&directory) {
+                    Ok(Some(pyproject)) => {
+                        match ruff_workspace::resolver::resolve_root_settings(
+                            &pyproject,
+                            Relativity::Parent,
+                            &EditorConfigurationTransformer(editor_settings, root),
+                        ) {
+                            Ok(settings) => {
+                                index.write().unwrap().insert(
+                                    directory,
+                                    Arc::new(RuffSettings {
+                                        path: Some(pyproject),
+                                        file_resolver: settings.file_resolver,
+                                        linter: settings.linter,
+                                        formatter: settings.formatter,
+                                    }),
+                                );
+                            }
+                            Err(err) => {
+                                let mut errors = errors.lock().unwrap();
+                                errors.push(Err(err));
+                            }
+                        }
+                    }
+                    Ok(None) => {}
+                    Err(err) => {
+                        let mut errors = errors.lock().unwrap();
+                        errors.push(Err(err));
+                    }
                 }
 
                 WalkState::Continue
             })
         });
 
+        let errors = errors.into_inner().unwrap();
+        if !errors.is_empty() {
+            tracing::error!(
+                "Errors while building the settings index in {}",
+                root.display()
+            );
+            for error in errors {
+                tracing::error!("{error}");
+            }
+        }
+
```

Comment:
I'm not sure if this is the way to go. We can't quit if we encounter an error as that would be a breaking change because currently we just continue to the next path.